tests/cpp/test_gpu_view.cpp

```diff
@@ -2584,39 +2584,77 @@ INSTANTIATE_TEST_SUITE_P(
+  auto tv1 = makeContigTensor(input_shape_wb.size(), DataType::Float);
+  auto tv2 = makeContigTensor(input_shape_wb.size(), DataType::Float);
   fusion->addInput(tv0);
-  auto tv1 = castOp(DataType::Float, tv0);
-  auto tv2 = reshape(tv1, input_shape, group_shape);
-  auto tv3 = sum(tv2, {-1, -2, -3});
-  auto tv4 = broadcast(tv3, {false, false, true, true, true});
-  auto tv5 = div(tv2, tv4);
-  auto tv6 = reshape(tv5, group_shape, input_shape);
-  auto tv7 = castOp(dtype, tv6);
-  fusion->addOutput(tv7);
+  fusion->addInput(tv1);
+  fusion->addInput(tv2);
+  auto tv3 = castOp(DataType::Float, tv0);
+  // reshape from {N, C, H, W} to {N, G, C / G, H, W}
+  auto tv4 = reshape(tv3, input_shape, group_shape);
+  // normalization
+  auto tv5 = sum(tv4, {-1, -2, -3});
+  auto tv6 = broadcast(tv5, {false, false, true, true, true});
+  auto tv7 = div(tv4, tv6);
+  // reshape scale and bias
+  auto tv8 = reshape(tv1, input_shape_wb, group_shape_wb);
+  auto tv9 = reshape(tv2, input_shape_wb, group_shape_wb);
+  // apply scale and bias
+  auto tv10 = broadcast(tv8, {true, false, false, true, true});
+  auto tv11 = broadcast(tv9, {true, false, false, true, true});
+  auto tv12 = mul(tv7, tv10);
+  auto tv13 = add(tv12, tv11);
+  auto tv14 = castOp(dtype, tv13);
+  // reshape back to input shape, segmented as a NoOp
+  auto tv15 = reshape(tv14, group_shape, input_shape);
+  fusion->addOutput(tv15);
 
   auto options =
       at::TensorOptions().dtype(data_type_to_aten(dtype)).device(at::kCUDA, 0);
+  auto options_wb = at::TensorOptions()
+                        .dtype(data_type_to_aten(DataType::Float))
+                        .device(at::kCUDA, 0);
   auto t0 = at::randn(input_shape, options);
+  auto tw = at::randn(input_shape_wb, options_wb);
+  auto tb = at::randn(input_shape_wb, options_wb);
   auto t1 = t0.reshape(group_shape).to(at::kFloat);
   auto t2 = t1.sum({-1, -2, -3}).unsqueeze(-1).unsqueeze(-1).unsqueeze(-1);
   auto t3 = t1 / t2;
-  auto ref = t3.reshape(input_shape).to(at::kHalf);
+  auto t4 = t3.reshape(input_shape);
+  auto t5 = tw.unsqueeze(0).unsqueeze(-1).unsqueeze(-1);
+  auto t6 = tb.unsqueeze(0).unsqueeze(-1).unsqueeze(-1);
+  auto t7 = t4.mul(t5).add(t6);
 
   FusionExecutorCache executor_cache(std::move(fusion));
-  auto cg_outputs = executor_cache.runFusionWithInputs({t0});
-  // should have 2 segmented groups
+  auto cg_outputs = executor_cache.runFusionWithInputs({t0, tw, tb});
   auto seg_groups =
       executor_cache.getMostRecentKernelRuntime()->fusionSegments()->groups();
-  EXPECT_EQ(seg_groups.size(), 2);
+
+  EXPECT_THAT(
+      seg_groups,
+      Contains(HeuristicIs(ScheduleHeuristic::InnerPersistent)).Times(1));
+  EXPECT_THAT(
+      seg_groups, Contains(HeuristicIs(ScheduleHeuristic::NoOp)).Times(1));
+
   testValidate(
-      executor_cache.fusion(), cg_outputs, {t0}, {ref}, __LINE__, __FILE__);
+      executor_cache.fusion(),
+      cg_outputs,
+      {t0, tw, tb},
+      {t7},
```

Comment:
I believe you can remove this line and let testValidate compute the reference output.